RFEM6_Adapter/CRUD/Create/Node.cs

```diff
@@ -42,25 +42,45 @@ private bool CreateCollection(IEnumerable<Node> bhNodes)
             foreach (Node bhNode in bhNodes)
             {
                 rfModel.node rfNode = bhNode.ToRFEM6();
+                m_Model.set_node(rfNode);
 
                 if (bhNode.Support != null)
                 {
-
-                    //TODO: 
-                    //API Call could possibly be reduced by combining the Node and Constraing6DOF Push...Problem, RFEM6 Needs Refernces in both Directions Nodes<->Nodal Support but BHoM has only a reference in o
-                    var rfSupport = m_Model.get_nodal_support(rfNode.support);
-                    HashSet<int> collectionOFSupporNo = rfSupport.nodes.ToHashSet();
-                    collectionOFSupporNo.Add(rfNode.no);
-                    rfSupport.nodes = collectionOFSupporNo.ToArray();
-                    m_Model.set_nodal_support(rfSupport);
+                    rfModel.object_with_children[] numbers = m_Model.get_all_object_numbers_by_type(rfModel.object_types.E_OBJECT_TYPE_NODAL_SUPPORT);
+                    List<rfModel.nodal_support> foundSupports = numbers.ToList().Select(n => m_Model.get_nodal_support(n.no)).ToList();
+                    var foundRFNodalSupport = foundSupports.Where(s => ComparerRFEMSupportAndBHoMConstraint(s, bhNode.Support)).FirstOrDefault();
+                    foundRFNodalSupport.nodes = foundRFNodalSupport.nodes.Append(rfNode.no).ToArray();
+                    m_Model.set_nodal_support(foundRFNodalSupport);
 
                 }
 
-                m_Model.set_node(rfNode);
             }
             return true;
         }
 
 
+        private static bool ComparerRFEMSupportAndBHoMConstraint(rfModel.nodal_support rfSupport, Constraint6DOF bhConstraint)
+        {
+
+            if (!tranlate(rfSupport.spring_x).Equals(bhConstraint.TranslationX)) { return false; }
```

Comment:
Please put all commands from an `if` onto its own line - no need for brackets if only a single line return is used.